internal/core/auth.go

```diff
@@ -0,0 +1,76 @@
+package core
+
+import (
+	"fmt"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/google/uuid"
+	"net/http"
+	"time"
+)
+
+type Claims struct {
+	jwt.RegisteredClaims
+	UserID string
+}
+
+const TokenExp = time.Hour * 1
+const SecretKey = "secret"
+const CookieName = "token"
+
+func GetUserToken(w http.ResponseWriter, r *http.Request) (string, error) {
+	var (
+		cookie *http.Cookie
+		err    error
+	)
+
+	cookie, _ = r.Cookie(CookieName)
+	if cookie == nil {
+		cookie, err = generateCookie()
+		if err != nil {
+			return "", fmt.Errorf("GetUserToken: failed to generate cookie, %s", err)
+		}
+		http.SetCookie(w, cookie)
+	}
+
+	claims := &Claims{}
+	token, err := jwt.ParseWithClaims(cookie.Value, claims,
+		func(t *jwt.Token) (interface{}, error) {
+			if _, ok := t.Method.(*jwt.SigningMethodHMAC); !ok {
+				return nil, err
+			}
+			return []byte(SecretKey), nil
+		})
+	if err != nil {
```

Comment:
```suggestion
	if err != nil  ||
	!token.Valid  {
```
можно немного сократить